controllers/topics/index.js

```diff
@@ -155,21 +156,49 @@ const followTopicV2 = async (req, res) => {
     const {name} = req.body;
     const {token, userId} = req;
 
+    // logic get user sign and anonymous
+    let secondDetailUser;
+    let secondDetailUserId;
+    const detailTokenUser = await UsersFunction.findUserById(User, userId);
+    if (!detailTokenUser.is_anonymous) {
+      secondDetailUser = await UsersFunction.findAnonymousUserId(User, userId);
+    } else {
+      secondDetailUserId = await UsersFunction.findSignedUserId(User, userId);
+      secondDetailUser = await UsersFunction.findUserById(User, secondDetailUserId);
+    }
+
     TopicValidator.validatePutTopicFollow({name});
 
+    //Logic get topic
     const topicService = new TopicService(Topics);
     const topic = await topicService.getTopicByName(name);
     const {topic_id} = topic;
     const userTopicService = new UserTopicService(UserTopic, UserTopicHistory);
-    const result = await userTopicService.followTopic(userId, topic_id);
 
-    const message = await _afterPutTopic(result, token, userId, name);
+    //get follow status
+    const [getTokenUserStatus, getSecondUserStatus] = await Promise.all([
+      userTopicService.getFollowTopicStatus(detailTokenUser.user_id, topic_id),
+      userTopicService.getFollowTopicStatus(secondDetailUser.user_id, topic_id)
+    ]);
+
+    let data = [];
+    if (getTokenUserStatus) {
+      await userTopicService.unfollowTopic(detailTokenUser.user_id, topic_id);
+      data.push(await _afterPutTopic(true, token, detailTokenUser.user_id, name));
+    } else {
+      await userTopicService.followTopic(detailTokenUser.user_id, topic_id);
+      data.push(await _afterPutTopic(false, token, detailTokenUser.user_id, name));
+
+      if (getSecondUserStatus) {
+        await userTopicService.unfollowTopic(secondDetailUser.user_id, topic_id);
+        data.push(await _afterPutTopic(true, token, secondDetailUser.user_id, name));
+      }
+    }
 
     return res.status(200).json({
       status: 'success',
       code: 200,
-      data: !result,
-      message
+      data: data
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for handling follow/unfollow operations seems to be correct. However, there is a potential issue with error handling. If an error occurs during the `unfollowTopic` or `followTopic` calls (lines 186, 189, and 193), it will not be caught until line 202, which means that the function might attempt to perform further operations even after an error has occurred. This could lead to inconsistent data states. Consider wrapping each of these calls in a try-catch block to handle errors immediately when they occur.

```diff
    if (getTokenUserStatus) {
      try {
        await userTopicService.unfollowTopic(detailTokenUser.user_id, topic_id);
        data.push(await _afterPutTopic(true, token, detailTokenUser.user_id, name));
      } catch (error) {
        // Handle error here
      }
    } else {
      try {
        await userTopicService.followTopic(detailTokenUser.user_id, topic_id);
        data.push(await _afterPutTopic(false, token, detailTokenUser.user_id, name));
      } catch (error) {
        // Handle error here
      }

      if (getSecondUserStatus) {
        try {
          await userTopicService.unfollowTopic(secondDetailUser.user_id, topic_id);
          data.push(await _afterPutTopic(true, token, secondDetailUser.user_id, name));
        } catch (error) {
          // Handle error here
        }
      }
    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->